### ISSUE TITLE
Fix disjoint sets of properties

### ISSUE THREAD

**Issue by Tritlo** [Tritlo/Endemic#60]
One issue is that we're trying to fix all the properties at once, which might be too much. One idea is to find disjoint sets of properties: i.e. if two properties do not mention any of the same identifiers, they are considered disjoint. Then we could fix one set of disjoint properties to generate a fix for those properties and then (hopefully) merge those fixes.

**Comment by Tritlo** [comment]
Finding fixes for these disjoint sets could also be done by a different type of search, maybe exhaustive or random. Then we could merge those solutions together with genetic search.

**Comment by lapplislazuli** [comment]
shouldn't be this inherently in the fitness function? 

Or do you mean that the user declares two sets of properties and the search tries to do a multi-objective optimization?

**Comment by Tritlo** [comment]
No, we can infer the disjoint sets ourselves, so the user doesn't do anything. But the search itself would be doing a multi-objective search which is (hopefully) an easier problem to solve.

**Comment by lapplislazuli** [comment]
I think its worth exploring, but that is actually another research question 😄 

I see the next paper coming: "Grouped Spectra Fault Localization - The future of Program Repair". 

Shall we mark it as Backlog for now?
